src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -138,6 +132,30 @@ Value mapToLowerScalarOp<ONNXCoshOp>(Operation *op, ArrayRef<Type> result_types,
   return result;
 }
 
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXTanhOp
+//===----------------------------------------------------------------------===//
+template <>
+Value mapToLowerScalarOp<ONNXTanhOp>(Operation *op, ArrayRef<Type> result_types,
+                                     ArrayRef<Value> operands,
+                                     ConversionPatternRewriter &rewriter) {
+  // ONNXTanhOp(%X) = DivFOp(SubFOp(ExpOp(%X), ExpOp(NegFOp(%X))),
+  //                         AddFOp(ExpOp(%X), ExpOp(NegFOp(%X))))
+  auto loc = op->getLoc();
+  Value operand = operands[0];
```

Comment:
At some point, we can use scalar or `std::pair` to represent these scalar operands; then we'll have c++ compiler make sure for us that we have the right number of scalar operands as input to this mapToLowerScalarOp function.